lib/qs/qs_runner.rb

```diff
@@ -17,6 +32,18 @@ def run_callbacks(callbacks)
       callbacks.each{ |proc| self.handler.instance_eval(&proc) }
     end
 
+    module OptionalTimeout
+      def self.after(timeout, &block)
```

Comment:
@jcredding I think I prefer `self.new` here.  the "after" name makes me think it should do this block after the time value has passed.

```ruby
OptionalTimeout.new(self.timeout) do
  # ...
end
```